python/ray/serve/_private/deployment_state.py

```diff
@@ -2034,14 +2065,23 @@ def _stop_replicas_on_draining_nodes(self):
             else:
                 self._replicas.add(replica.actor_details.state, replica)
 
-    def update(self) -> DeploymentStateUpdateResult:
+    def update(
+        self, target_capacity: Optional[float] = None
```

Comment:
I much prefer moving in the direction of these functions being pure. I found it quite hard to follow the existing logic between autoscaling and the rest of the reconciliation because `.autoscale()` was overwriting the `target_info`, which I assumed was immutable and represented the target config.

In general testing and reasoning about the code is easier for pure functions.

Reply:
`self._target_state.num_replicas` is now only being used/adjusted at (1) the top level when calculating the adjusted version and (2) in `autoscale()` (which I think we should fix)